internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -175,7 +179,15 @@ func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, nil
 	}
 
-	gitPollerWasExecuted, _ := r.checkPollingTask(ctx, gitrepo)
+	commitBefore := gitrepo.Status.Commit
+	gitPollerWasExecuted, err := r.checkPollingTask(ctx, gitrepo)
+	if err != nil {
+		r.Recorder.Event(gitrepo, fleetevent.Warning, "FailedToCheckCommit", err.Error())
+	} else {
+		if gitPollerWasExecuted && commitBefore != gitrepo.Status.Commit {
+			r.Recorder.Event(gitrepo, fleetevent.Normal, "GotNewCommit", gitrepo.Status.Commit)
+		}
+	}
```

Comment:
nit: we could save a couple of lines and an indentation level:
```suggestion
	} else if gitPollerWasExecuted && commitBefore != gitrepo.Status.Commit {
	        r.Recorder.Event(gitrepo, fleetevent.Normal, "GotNewCommit", gitrepo.Status.Commit)
	}
```

Reply:
changed as suggested